libs/ssl-config/src/main/java/org/elasticsearch/common/ssl/SslConfigurationLoader.java

```diff
@@ -281,21 +287,22 @@ public SslConfiguration load(Path basePath) {
         final SslVerificationMode verificationMode = resolveSetting(VERIFICATION_MODE, SslVerificationMode::parse, defaultVerificationMode);
         final SslClientAuthenticationMode clientAuth = resolveSetting(CLIENT_AUTH, SslClientAuthenticationMode::parse, defaultClientAuth);
 
-        final SslTrustConfig trustConfig = buildTrustConfig(basePath, verificationMode);
         final SslKeyConfig keyConfig = buildKeyConfig(basePath);
+        final SslTrustConfig trustConfig = buildTrustConfig(basePath, verificationMode, keyConfig);
 
         if (protocols == null || protocols.isEmpty()) {
             throw new SslConfigException("no protocols configured in [" + settingPrefix + PROTOCOLS + "]");
         }
         if (ciphers == null || ciphers.isEmpty()) {
             throw new SslConfigException("no cipher suites configured in [" + settingPrefix + CIPHERS + "]");
         }
-        return new SslConfiguration(trustConfig, keyConfig, verificationMode, clientAuth, ciphers, protocols);
+        final boolean isExplicitlyConfigured = hasSettings(settingPrefix);
+        return new SslConfiguration(isExplicitlyConfigured, trustConfig, keyConfig, verificationMode, clientAuth, ciphers, protocols);
     }
 
-    private SslTrustConfig buildTrustConfig(Path basePath, SslVerificationMode verificationMode) {
-        final List<Path> certificateAuthorities = resolveListSetting(CERTIFICATE_AUTHORITIES, basePath::resolve, null);
-        final Path trustStorePath = resolveSetting(TRUSTSTORE_PATH, basePath::resolve, null);
+    protected SslTrustConfig buildTrustConfig(Path basePath, SslVerificationMode verificationMode, SslKeyConfig keyConfig) {
+        final List<String> certificateAuthorities = resolveListSetting(CERTIFICATE_AUTHORITIES, Function.identity(), null);
+        final String trustStorePath = resolveSetting(TRUSTSTORE_PATH, Function.identity(), null);
```

Comment:
There's a number of changes that switch from `Path` to `String`. That's needed for the SSL certificates API. It should report the "path" as configured in the settings, not the path as on disk.

That is it reports `http.p12` not `${ES_HOME}/config/http.p12`.
So, all of these config classes needed to change to use `String` instead of `Path` and resolve the file at the last possible moment.

I considered creating a new class just for this (and still can) but on balance is seemed like passing around a new `ConfigPath` object would be more confusing.